src/ManagedServices/ManagedServices/Commands/RemoveAzureRmManagedServcicesAssignment.cs

```diff
@@ -0,0 +1,71 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Commands
+{
+    [Cmdlet(
+        VerbsCommon.Remove,
+        Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesAssignment",
+        DefaultParameterSetName = DefaultParameterSet,
+        SupportsShouldProcess = true), OutputType(typeof(PSRegistrationDefinition))]
+    public class RemoveAzureRmManagedServcicesAssignment : ManagedServicesCmdletBase
+    {
+        protected const string DefaultParameterSet = "Default";
+        protected const string InputObjectParameterSet = "InputObject";
+
+        [Parameter(Position = 0, ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The subscription scope.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The registration assignment Id.")]
```

Comment:
So, in this case, because you are either using Scope + Id to define the resource assign,ment, or you are piping from Get-AzResource, you should have separate parameters for Id and ResourceId, with Id being in the defualt parameter set, and ResourceId being in its own separate parameter set (and teking value from pipeline by property name)